scheduler/scheduler.go

```diff
@@ -11,16 +12,27 @@ import (
 // be handled via a config file in a prod system.
 const (
 	grpcServerAddr    = "127.0.0.1:50000"
-	grpcServerUseTLS  = false
-	grpcServerCrtFile = "server.pem"
-	grpcServerKeyFile = "server.key"
+	grpcServerUseTLS  = true
+	grpcServerCrtFile = "scheduler.crt"
+	grpcServerKeyFile = "scheduler.key"
+
+	defAPIKey = "default-api-key"
```

Comment:
I really don't like having a default api key here. 

If this project was to build an MVP and to give it to a customer, they could easily miss the api key flag, and completely violate the security model by accident. So I generally recommend, even for toy projects, to try as much as possible to avoid embedding secrets (except as part of unit tests can be acceptable, but even those can be generated when running tests).

Reply:
I was exiting when the API key was not provided via flag here. But I decided to change it for testing purposes. Definitely would not leave it like this on a prod system